test-suite/exonum-py-tests/suite/process_manager.py

```diff
@@ -0,0 +1,84 @@
+"""Tools for managing subprocesses."""
+from enum import Enum, auto as enum_auto
+import subprocess
+from typing import NamedTuple, Optional
+from threading import Thread
+import os
+import signal
+
+
+class ProcessExitResult(Enum):
+    """Result of the process termination."""
+
+    # Process exited succesfully.
+    Ok = enum_auto()
+    # Process did not exit succesfully and was killed.
+    Killed = enum_auto()
+
+
+class ProcessOutput(NamedTuple):
+    """Outputs collected during process execution."""
+
+    exit_result: ProcessExitResult
+    exit_code: int
+    stdout: str
+    stderr: str
+
+
+class ProcessManager:
+    """ProcessManager is the entity capable of the running the process
+    in the separate thread, joining it and collectiong outputs."""
+
+    def __init__(self, command: str):
+        self._thread_handle = Thread(target=self._start_process)
```

Comment:
Well, yeah, but since we are launching the `exonum` in the shell, a separate process will be created anyway. This thread is only a wrapper around the launched process, and I don't think it should be a process itself. For example, here we do share some state between spawned thread and original one (`self._output`) and GIL is our best friend to not bother ourselves with synchronization.

Reply:
Ouh, OK. I seem to forget about `self._output`. `self._start_process` was a bit misleading. Maybe a `self._start_thread` instead?

Reply:
It's a bit complicated, since we're starting a thread which starts a process via subprocess `¯\_(ツ)_/¯`

I'm not sure about the naming, but I think since this entity is called `ProcessManager` and it's all about the spawned process and its outputs, I think that process at least should be mentioned.

How about `_spawn_subprocess` or `_spawn_subprocess_in_thread`?

Reply:
Or even simplier: just `_start`

Reply:
I guess `_start_subprocess` will be fine.